Moonshot/procedural_map_generation/Room.gd

```diff
@@ -10,18 +10,23 @@ var room_template_name
 var room_scene
 var room_instance
 var room_instanced = null
-
-
+var level_no
+var room_size_x
+var room_size_y
+var difficulty_rating 
+# Controls how quickly the levels get harder as the levels go up. set between 0.5 - 1
+var scaling_factor = 0.75
 # Type of room that need to be considered
 const ROOM_TYPES : Array = ["Boss", "Reward", "Shop", "Challenge", "Path", "CurrentLocation"]
```

Comment:
Looking at this list with an eye on the time. How do we feel about creating this much variety?

Reply:
Fair comment, and I think it is worth a discussion. I personally think that there will be very little variety in the 'shop' and 'reward' rooms as they are always the edge rooms with one door, and will not have enemies in. I think we need one of the shop/reward otherwise there is no benefit from exploration.